core/src/main/java/io/questdb/cairo/CairoEngine.java

```diff
@@ -1407,18 +1424,21 @@ private void buildMatViewGraph() {
                         }
 
                         final MatViewRefreshState state = matViewGraph.addView(matViewDefinition);
-                        final boolean isMatViewStateExists = TableUtils.isMatViewStateFileExists(configuration, path, tableToken.getDirName());
-                        path.trimTo(pathLen).concat(tableToken.getDirName()).concat(MatViewRefreshState.MAT_VIEW_STATE_FILE_NAME);
-                        if (isMatViewStateExists) {
-                            reader.of(path.$());
-                            MatViewRefreshState.readFrom(reader, state);
-                        } else {
-                            blockFileWriter.of(path.$());
-                            MatViewRefreshState.append(state, blockFileWriter);
-                        }
+                        // can be null if the graph has no-op implementation
+                        if (state != null) {
+                            final boolean isMatViewStateExists = TableUtils.isMatViewStateFileExists(configuration, path, tableToken.getDirName());
+                            path.trimTo(pathLen).concat(tableToken.getDirName()).concat(MatViewRefreshState.MAT_VIEW_STATE_FILE_NAME);
+                            if (isMatViewStateExists) {
+                                reader.of(path.$());
```

Comment:
this is a startup activity, lets not use thread-local path here, it is error prone

Reply:
Changed in d914617fa373d60e9a989fa9ddbc4a9839b2ea9a

Reply:
Changed in d914617fa373d60e9a989fa9ddbc4a9839b2ea9a